File: petclinic-data/src/main/java/xyz/haff/petclinic/models/Owner.java
package xyz.haff.petclinic.models;

import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.Cascade;

import javax.persistence.*;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@ToString(callSuper = true)
@SuperBuilder
public class Owner extends Person {

    @ToString.Exclude
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "owner")
    private Set<Pet> pets;
}

File: petclinic-data/src/main/kotlin/xyz/haff/petclinic/models/Owner.kt
package xyz.haff.petclinic.models

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import java.util.*

data class Owner @JvmOverloads constructor(
    val baseEntity: BaseEntity = BaseEntity(),
    val personalData: PersonalData
): Entity by baseEntity {
}
